### DISCUSSION TITLE
Samba & Wifibox

### DISCUSSION THREAD

**Discussion by Thomasst24** [pgj/freebsd-wifibox#140] [Q&A] [answered]
Hi there. Is it possible to use a samba connection with wifibox? Tried  & didn't work.
So i'm back to my wired connection

**Comment by pgj** [comment, marked as answer]
I believe it should be possible.  I do not employ Samba where I use Wifibox so I do not have direct experience with this.  I assume that the issue is caused by `iptables` on the guest.

By default, `iptables` does not forward the incoming traffic to the FreeBSD host due to its NAT-style setup.  But its configuration could be changed to do that for designated ports, e.g. the ones that Samba uses.  These should be UDP/137 (NetBIOS name service), UDP/138 (NetBIOS datagram), TCP/139 (NetBIOS session), TCP/445 (Active Directory, Windows shares), and UDP/445 (SMB file sharing).

You can find the `iptables` rules in `${LOCALBASE}/etc/wifibox/appliance/iptables` (where `${LOCALBASE}` is usually `/usr/local`).  There you would have to add something like the following lines to the `*nat` section:

```
[0:0] -A PREROUTING -p tcp -m multiport --dports 139,445 -j DNAT --to-destination 10.0.0.2
[0:0] -A PREROUTING -p udp --dport 137:138 -j DNAT --to-destination 10.0.0.2
```

where `10.0.0.2` is the IP address of the `wifibox0` interface on the FreeBSD host.  Remember to either restart the guest after this change:

```console
# wifibox restart guest
```

Or enter the console and restart `iptables` itself:

```console
# wifibox console
<login as root>
# services iptables restart
```

**Reply by Thomasst24** [reply]
Worked. Thanks
